File: MusicPlayer.v2/app/src/main/java/com/example/musicplayerv2/AppListDiffCallback.java
package com.example.musicplayerv2;

import android.support.v7.util.DiffUtil;

/**
 * Created by ${Aina} on 11.11.2018.
 */
class AppListDiffCallback extends DiffUtil.ItemCallback<Song> {
    @Override
    public boolean areItemsTheSame(Song oldItem, Song newItem) {
        return oldItem.getId() == newItem.getId();
    }

    @Override
    public boolean areContentsTheSame(Song oldItem, Song newItem) {
        return oldItem.getArtist().equals(newItem.getArtist());
    }
}

File: MusicPlayer.v2Kotlin/app/src/main/java/com/example/musicplayerv2/AppListDiffCallback.kt
package com.example.musicplayerv2

import android.support.v7.util.DiffUtil

/**
 * Created by ${Aina} on 11.11.2018.
 */
internal class AppListDiffCallback : DiffUtil.ItemCallback<Song>() {
    override fun areItemsTheSame(oldItem: Song, newItem: Song): Boolean = oldItem.id == newItem.id

    override fun areContentsTheSame(oldItem: Song, newItem: Song): Boolean = oldItem.artist == newItem.artist
}
